ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -32,107 +36,260 @@ public class GltfModelExporter
                     }
+                    var node = AddMeshNode(exportedModel, scene, model,
+                        meshName, meshes[i].Mesh, model.GetSkeleton(i),
+                        skinName != null ? GetSkinPathFromModel(model, skinName) : null );
+
+                    if (node == null) continue;
+                    // Swap Rotate upright, scale inches to meters.
+                    node.WorldMatrix = transform * TRANSFORMSOURCETOGLTF;
                 }
-                else
+            }
+
+            foreach (var childEntityName in entityLump.GetChildEntityNames())
+            {
+                if (childEntityName == null) continue;
+                var childEntityLumpResource = FileLoader.LoadFile(childEntityName + "_c");
+                if (childEntityLumpResource == null) continue;
+
+                var childEntityLump = (VEntityLump)childEntityLumpResource.DataBlock;
+                LoadEntityMeshes(exportedModel, scene, childEntityLump);
+            }
+        }
+
+        private static string GetSkinPathFromModel(VModel model, string skinName)
+        {
+            var materialGroupForSkin = model.Data.GetArray<IKeyValueCollection>("m_materialGroups")
+                .ToList()
+                .Where(m => m.GetProperty<string>("m_name") == skinName)
+                .SingleOrDefault();
+
+            if (materialGroupForSkin == null) return null;
+
+            // Given these are at the model level, and otherwise pull materials from drawcalls
+            // on the mesh, not sure how they correlate if there's more than one here
+            // So just take the first one and hope for the best
+            return materialGroupForSkin.GetArray<string>("m_materials")[0];
+        }
+
+        private static Matrix4x4 CalculateTransformationMatrix(VEntityLump.Entity entity)
+        {
+            var scale = entity.GetProperty<string>("scales");
+            var position = entity.GetProperty<string>("origin");
+            var angles = entity.GetProperty<string>("angles");
+            if (scale == null || position == null || angles == null)
+            {
+                return default;
+            }
+
+            var scaleMatrix = Matrix4x4.CreateScale(ParseVector(scale));
+
+            var positionVector = ParseVector(position);
+            var positionMatrix = Matrix4x4.CreateTranslation(positionVector);
+
+            var pitchYawRoll = ParseVector(angles);
+            var rollMatrix = Matrix4x4.CreateRotationX(pitchYawRoll.Z * ((float)Math.PI / 180f)); // Roll
+            var pitchMatrix = Matrix4x4.CreateRotationY(pitchYawRoll.X * ((float)Math.PI / 180f)); // Pitch
+            var yawMatrix = Matrix4x4.CreateRotationZ(pitchYawRoll.Y * ((float)Math.PI / 180f)); // Yaw
+
+            var rotationMatrix = rollMatrix * pitchMatrix * yawMatrix;
+            return scaleMatrix * rotationMatrix * positionMatrix;
+        }
+
+        private static Vector3 ParseVector(string input)
```

Comment:
Did this and the function above get copied from the renderer? Would be better to split them into a separate class instead of duplcating the code.

Reply:
It did. I can move those into a separate class in the ValveResourceFormat project.